### PR TITLE
feat(rocm): honour extra compile-flag env vars on the HIP JIT path

### PR DESCRIPTION
**Pull request by demandal25**
## Summary

The CUDA path has read `FLASHINFER_EXTRA_CFLAGS` and `FLASHINFER_EXTRA_CUDAFLAGS` since [`cpp_ext.py:189`](https://github.com/AMD-Ecosystem/flashinfer/blob/amd-integration/flashinfer/jit/cpp_ext.py#L189); the HIP path had a hook for link flags only, so there was no way to add a compile flag to a JIT build without editing the generator. This mirrors both, and fixes two ways the host compile rule mishandles offload flags.

### What changed

- **`flashinfer/jit/cpp_ext_hip.py`** — reads `FLASHINFER_EXTRA_CFLAGS` and `FLASHINFER_EXTRA_CUDAFLAGS`; one `shlex`-with-fallback helper now also replaces the open-coded copy in the existing `FLASHINFER_EXTRA_LDFLAGS` hook. Hardens `_for_host()`. Adds `FLASHINFER_OWN_HEADERS_NON_SYSTEM`.
- **`tests/rocm_tests/test_jit_flag_hooks.py`** — 28 tests over the flag helpers and the rendered ninja. Stubs torch, so no GPU.
- **`.github/workflows/arch-caps-conformance.yml`** — runs that suite. It stubs torch precisely so it can run on a CPU runner, and no other lane collects `tests/rocm_tests/`, so without this step the stubbing bought nothing.
- **`CLAUDE.md`** — the three new variables in the commands table.

### Why CFLAGS is host-only

This is the one place the CUDA and HIP generators could not be written alike, and it is worth stating because the failure would have been silent.

The `hip_compile` rule ends with `$cflags`:

```
command = $amdclang -xhip -MD -MF $out.d $cuda_cflags -c $in -o $out $cuda_post_cflags $cflags
```

So a flag appended to `cflags` reaches **device** codegen too, and being last on the command line it outranks both the `-O3` injected by `CompilationContext.get_hipcc_flags_list()` and anything in `FLASHINFER_EXTRA_CUDAFLAGS`. `FLASHINFER_EXTRA_CFLAGS=-O0` would have silently de-optimised every kernel, with `FLASHINFER_EXTRA_CUDAFLAGS=-O3` unable to win it back. The CUDA `cuda_compile` rule does not reference `$cflags`, so there the variable is host-only. It is host-only here too — applied to `host_cflags` after `_for_host()`, leaving the pre-existing `$cflags` tail untouched.

### `_for_host()` hardening

`--offload-arch` was matched by prefix and consumed only its own token. Two consequences, both measured against ROCm 6.4.1:

| Input | Before | After |
|---|---|---|
| `--offload-arch gfx942` (separate-arg form) | strands `gfx942` → `c++: error: gfx942: linker input file not found` | both tokens dropped |
| `--offload-arch-list=x` | silently dropped by prefix match | preserved |

`-Xarch_host X` is unwrapped to `X` and `-Xarch_device X` dropped, because the `compile` rule is not an offload compile: `c++ -Xarch_host -DFOO -c t.cc` gives `error: unrecognized command-line option '-Xarch_host'`. `aiter_loader.cc` and `spdlog.py`'s `logging.cc` are the sources on that rule. Every one of these failures reads as a compiler problem rather than a flag problem, which is why they are worth a test each.

### `FLASHINFER_OWN_HEADERS_NON_SYSTEM`

`-isystem` exists to keep third-party warnings quiet, but it applies the same silence to our own headers — `include/` and `csrc_rocm/` are currently invisible to `-Wall -Wextra`. Setting this to `1` emits them as `-I` instead. Measured on a header with an unused variable and an assignment-in-condition:

| Include flag | Warnings from `-Wall -Wextra` |
|---|---|
| `-isystem` | 0 |
| `-I` | 4 |

Third-party include dirs are unaffected and stay `-isystem`. Instrumentation passes are suppressed in system headers for the same reason, so this is also what a sanitiser or profile build needs.

## Test plan

- [x] `pytest --noconftest tests/rocm_tests/test_jit_flag_hooks.py` — 28 passed, no GPU, no torch.
- [x] Every new test A/B'd against the unfixed code; each fails there.
- [x] Rendered the full ninja for a `.cu` + `.cc` pair with all three variables unset, before and after: **byte-identical**, so the default build is unchanged.
- [x] `-Xarch_host` and `--offload-arch` claims reproduced against ROCm 6.4.1 `c++`/`amdclang++` rather than reasoned about.
- [x] `pre-commit run -a`.

Arch coverage: the change is pure string manipulation over flag lists with no arch branching, so gfx942 and gfx950 take the identical path — no per-arch behaviour to verify. No JIT build was run on either; the rendered-ninja comparison above is what stands in for it.

### CODE REVIEW

File: flashinfer/jit/cpp_ext_hip.py [hunk 1]
```diff
@@ -117,8 +164,14 @@ def generate_ninja_build_for_op(
...
-    # host_cflags: strip flags the host c++ compiler doesn't understand (--offload-arch)
-    host_cflags = [f for f in cflags if not f.startswith("--offload-arch")]
+    # Mirrors FLASHINFER_EXTRA_CFLAGS/CUDAFLAGS on the CUDA path (cpp_ext.py);
+    # the HIP path previously had a hook for link flags only.
+    #
+    # CFLAGS lands here, on the host rule alone, rather than in `cflags`: the
+    # hip_compile rule below ends with `$cflags`, so anything put there would
+    # also reach device codegen and, being last, would outrank both -O3 and
+    # FLASHINFER_EXTRA_CUDAFLAGS. On CUDA the var is host-only; keep it so.
+    host_cflags = _for_host(cflags) + _env_flags("FLASHINFER_EXTRA_CFLAGS")
```
[the remark this answers is not in the document]
Comment (demandal25) [reply flashinfer/jit/cpp_ext_hip.py]: Right, and this one was mine — scoping CFLAGS to the host rule put it *after* `_for_host` instead of through it, bypassing the exact hardening this PR adds. Fixed in 4c658c91d: `_for_host(cflags + _env_flags(...))`, which keeps the flags out of `cflags` (device codegen untouched) while normalising them. New test fails against the previous form.

File: flashinfer/jit/cpp_ext_hip.py [hunk 2]
```diff
@@ -51,6 +100,11 @@ def generate_ninja_build_for_op(
         "$torch_home/include",
         "$torch_home/include/torch/csrc/api/include",
         "$cuda_home/include",
```
[the remark this answers is not in the document]
Comment (demandal25) [reply flashinfer/jit/cpp_ext_hip.py]: Correct that `cuda_home` is never defined, but declining: it is pre-existing and only a context line in this diff, and it is inert. `$cuda_home/include` expands to `/include`, which does not exist, and clang ignores a nonexistent `-isystem` directory silently:

```
$ ls -d /include
ls: cannot access '/include': No such file or directory
$ clang++ -Wall -Wextra -isystem /include -c m.cc -o /dev/null
(no output, rc=0)
```

Worth removing, but not in a PR about env-var hooks — the line predates it and removing it changes a path this PR does not otherwise touch.

File: tests/rocm_tests/test_jit_flag_hooks.py [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+            get_rocm_home=lambda: "/stub/rocm"
+        ),
+        # The module does `from . import env`, so it needs a real package chain.
+        "flashinfer": types.ModuleType("flashinfer"),
+        "flashinfer.jit": types.ModuleType("flashinfer.jit"),
+        "flashinfer.jit.env": types.SimpleNamespace(
+            FLASHINFER_INCLUDE_DIR=Path("/stub/include"),
+            FLASHINFER_CSRC_DIR=Path("/stub/csrc_rocm"),
+        ),
+    }
```
[the remark this answers is not in the document]
Comment (demandal25) [reply tests/rocm_tests/test_jit_flag_hooks.py]: Declining — the 29 tests in this file all go through `_load_module()` and pass, which already disproves the `ModuleNotFoundError`. The mechanism: `__path__` is only consulted when a submodule has to be *found* on disk. These are preloaded into `sys.modules`, so both import forms short-circuit there.

Reproduced with parents deliberately lacking `__path__`, exactly as the stubs are:

```
torch                  has __path__: False
torch.utils            has __path__: False
flashinfer             has __path__: False
flashinfer.jit         has __path__: False
from torch.utils.cpp_extension import ... -> /stub
from . import env -> 1
```

`test_arch_caps_hip.py` sets `__path__` because it stubs a parent *without* preloading the submodule, so there the lookup does reach the finder. Different shape, not an inconsistency.
